op-batcher/batcher/driver.go

```diff
@@ -665,6 +669,23 @@ func (l *BatchSubmitter) l1Tip(ctx context.Context) (eth.L1BlockRef, error) {
 	return eth.InfoToL1BlockRef(eth.HeaderBlockInfo(head)), nil
 }
 
+func (l *BatchSubmitter) checkTxpool(queue *txmgr.Queue[txRef], receiptsCh chan txmgr.TxReceipt[txRef]) bool {
+	l.txpoolMutex.Lock()
+	if l.txpoolState == TxpoolBlocked {
+		// txpoolState is set to Blocked only if Send() is returning
+		// ErrAlreadyReserved. In this case, the TxMgr nonce should be reset to nil,
+		// allowing us to send a cancellation transaction.
+		l.txpoolState = TxpoolCancelPending
+		isBlob := l.txpoolBlockedBlob
+		l.txpoolMutex.Unlock()
+		l.cancelBlockingTx(queue, receiptsCh, isBlob)
+		return false
+	}
+	r := l.txpoolState == TxpoolGood
+	l.txpoolMutex.Unlock()
```

Comment:
I see that you don't call the mutex unlocking in a `defer` to release the lock before calling `l.cancelBlockingTx`. Would it actually hurt to hold the lock when calling it? It's only enqueueing the tx, not blocking on sending it.

Reply:
Unfortunately because we cap the number of parallel sends, queuing actually *can* block.  This blocking is also part of why we can get get stuck in the state publishing loop for a very long time.

Reply:
FWIW this has bitten me before.. I wonder if we should rename queueTx so it's clear it can block.  I did update the godoc for queue.Send() the last time this bit me to clarify it does block, but I notice queueTx here (which calls queue.Send()) doesn't have any such comment.

Reply:
went ahead and renamed it to `sendTx` and updated the godoc!

Reply:
Got it, I suspected this... thanks!